### DISCUSSION TITLE
binary_sensor with device_class: for unavailable entities always "on-icon" is used

### DISCUSSION THREAD

**Discussion by ildar170975** [home-assistant/frontend#15882] [Dashboard]
Every binary sensor with a specified device_class has "on-icon" & "off-icon".
But when an entity is `unavailable` - then currently always "on-icon" is used.
In many cases it is confusing, for instance:

![изображение](https://user-images.githubusercontent.com/71872483/226257311-b8d5845e-98e1-4420-98e9-416b267ea4ba.png)
![изображение](https://user-images.githubusercontent.com/71872483/226257336-88a305c2-c421-4b10-ad35-ca7e34ac01af.png)

Imho for `unavailable` entities the "off-icon" should be used.
From other side, the currently implemented logics may be considered as "this entity is unavailable, so it is of grey color; and we use an icon demonstrating a purpose of this entity" (i.e. icon is not related to the current state, it is just to demonstrate this particular `device_class`).

**Comment by karwosts** [comment]
I think this is implemented now.

Anyway, icons are now chosen by core. So perhaps not a frontend issue. 

<img width="473" height="57" alt="image" src="https://github.com/user-attachments/assets/6697b834-e425-4530-97f4-31175eec6f96" />
<img width="482" height="55" alt="image" src="https://github.com/user-attachments/assets/d432c184-83fe-49c5-8714-51ef19b321b0" />
<img width="482" height="58" alt="image" src="https://github.com/user-attachments/assets/7c96a44e-2588-46fb-9112-0e59ded81aa8" />

Example icon definition:

```
    "cold": {
      "default": "mdi:thermometer",
      "state": {
        "on": "mdi:snowflake"
      }
    },
 ```
